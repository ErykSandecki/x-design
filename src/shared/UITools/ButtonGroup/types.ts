// types
import { TTooltipProps } from '../../UI/Tooltip/Tooltip';
import { TIconProps } from '../../UI/Icon/Icon';

export type TButtonGroup = {
  disabled?: boolean;
  name: TIconProps['name'];
  onClick: TFunc;
  tooltip?: Omit<TTooltipProps, 'children'>;
};
