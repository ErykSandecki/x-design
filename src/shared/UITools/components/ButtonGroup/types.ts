// types
import { TTooltipProps } from '../../../UI/components/Tooltip/Tooltip';
import { TIconProps } from '../../../UI/components/Icon/Icon';

export type TButtonGroup = {
  disabled?: boolean;
  name: TIconProps['name'];
  onClick: TFunc;
  tooltip?: Omit<TTooltipProps, 'children'>;
};
