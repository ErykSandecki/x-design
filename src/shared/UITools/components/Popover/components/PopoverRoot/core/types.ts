export type TContext = {
  setSelected: TFunc<[boolean]>;
};
