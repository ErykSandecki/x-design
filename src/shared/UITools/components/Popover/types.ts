// types
import { TIconProps } from 'shared/UI/components/Icon/Icon';

export type TPopoverData = {
  icon?: TIconProps['name'];
  onClick?: () => void;
  selected?: boolean;
  text: string;
};

export type TPopoverSeparator = { separator: boolean };

export type TPopoverItem = TPopoverData | TPopoverSeparator;

export type TPopover = {
  data: Array<TPopoverData | TPopoverSeparator>;
};
