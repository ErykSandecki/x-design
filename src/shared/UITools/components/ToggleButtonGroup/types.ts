import { TIconProps } from 'shared/UI/components/Icon/Icon';
import { TToggleButtonProps } from './components/ToggleButton/ToggleButton';

export type TToggleButton<V> = Pick<TToggleButtonProps<V>, 'tooltip' | 'value'> & {
  icon: TIconProps['name'];
};

export type TToggleButtonGroupValue = string | Array<string>;
