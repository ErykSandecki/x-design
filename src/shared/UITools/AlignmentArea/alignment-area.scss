.AlignmentArea {
  @include theme('neutral-3', background-color);
  @include theme('neutral-3', border, '1px solid');

  align-items: center;
  border-radius: 5px;
  box-sizing: border-box;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(3, 1fr);
  height: 56px;
  justify-content: center;

  &--full-width {
    width: 100%;
  }
}
