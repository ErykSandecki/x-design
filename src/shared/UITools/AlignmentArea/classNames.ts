import { values } from 'lodash';

// types
import { AlignmentFlow } from 'types';

// utils
import { composeClassNames } from 'utils';

export const className = 'AlignmentArea';

export const classNames = composeClassNames(
  className,
  [className, 'fullWidth'] as const,
  ['option', 'selected'] as const,
  ['optionView', ...values(AlignmentFlow)] as const,
  ['indicator', 'selected'] as const,
);
