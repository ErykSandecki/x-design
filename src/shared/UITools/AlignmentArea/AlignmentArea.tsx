import { FC } from 'react';

// components
import AlignmentOption from './AlignmentOption/AlignmentOption';
import Box from '../../UI/Box/Box';

// hooks
import { useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// styles
import styles from './alignment-area.scss';

// types
import { AlignmentFlow, E2EAttribute } from 'types';
import { TE2EDataAttributeProps } from '../../E2EDataAttributes/E2EDataAttribute';

// utils
import { enumToArray } from 'utils';

export type TAlignmentAreaProps = {
  e2eValue?: TE2EDataAttributeProps['value'];
  fullWidth?: boolean;
  onClick: TFunc<[AlignmentFlow]>;
  value: AlignmentFlow;
};

export const AlignmentArea: FC<TAlignmentAreaProps> = ({ e2eValue = '', fullWidth = false, onClick, value }) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);

  return (
    <Box
      classes={{
        className: cx(className, classNamesWithTheme[className].name, [
          classNamesWithTheme[className].modificators.fullWidth,
          fullWidth,
        ]),
      }}
      e2eAttribute={E2EAttribute.alignmentArea}
      e2eValue={e2eValue}
    >
      {enumToArray<AlignmentFlow>(AlignmentFlow).map((alignmentFlow) => (
        <AlignmentOption
          alignmentFlow={alignmentFlow}
          isSelected={value === alignmentFlow}
          key={alignmentFlow}
          onClick={onClick}
        />
      ))}
    </Box>
  );
};

export default AlignmentArea;
