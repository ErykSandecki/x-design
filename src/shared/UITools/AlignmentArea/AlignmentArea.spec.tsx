import { fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';

// components
import AlignmentArea from './AlignmentArea';

// mocks
import 'test/mocks/sagas/allSagas';

// store
import { configureStore } from 'store/store';

// types
import { AlignmentLayout, E2EAttribute } from 'types';

// utils
import { customRender, getByE2EAttribute } from 'test';

const mockCallBack = jest.fn();

describe('AlignmentArea snapshots', () => {
  it('should render AlignmentArea', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = customRender(
      <Provider store={store}>
        <AlignmentArea onClick={mockCallBack} value={AlignmentLayout.bottomCenter} />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('AlignmentArea behaviors', () => {
  it('should trigger change alignment', () => {
    // mock
    const store = configureStore();

    // before
    const { container } = customRender(
      <Provider store={store}>
        <AlignmentArea onClick={mockCallBack} value={AlignmentLayout.bottomCenter} />
      </Provider>,
    );

    // action
    fireEvent.click(getByE2EAttribute(container, E2EAttribute.alignmentAreaOption, AlignmentLayout.bottomLeft));

    // result
    expect(mockCallBack.mock.calls[0][0]).toBe(AlignmentLayout.bottomLeft);
  });
});
