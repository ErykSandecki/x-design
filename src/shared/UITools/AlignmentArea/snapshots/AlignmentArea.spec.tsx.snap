// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlignmentArea snapshots should render AlignmentArea 1`] = `
<DocumentFragment>
  <div
    class="css-1jnxvbp AlignmentArea AlignmentArea AlignmentArea--light"
    data-test-alignment-area=""
  >
    <div
      class="css-1jnxvbp AlignmentOption AlignmentOption--light"
      data-test-alignment-area-option="topLeft"
      data-test-box=""
    >
      <div
        class="css-1jnxvbp AlignmentOption__optionView AlignmentOption__optionView--light AlignmentOption__optionView--topLeft AlignmentOption__optionView--topLeft--light"
        data-test-box=""
      >
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
      </div>
    </div>
    <div
      class="Tooltip Tooltip--light"
    >
      <div
        class="Tooltip__content Tooltip__content--light"
      >
        Align top left
        <div
          class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
          data-test-icon="carrot-down"
          height="14"
          name="CarrotDown"
          width="10"
        />
      </div>
    </div>
    <div
      class="css-1jnxvbp AlignmentOption AlignmentOption--light"
      data-test-alignment-area-option="topCenter"
      data-test-box=""
    >
      <div
        class="css-1jnxvbp AlignmentOption__optionView AlignmentOption__optionView--light AlignmentOption__optionView--topCenter AlignmentOption__optionView--topCenter--light"
        data-test-box=""
      >
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
      </div>
    </div>
    <div
      class="Tooltip Tooltip--light"
    >
      <div
        class="Tooltip__content Tooltip__content--light"
      >
        Align top center
        <div
          class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
          data-test-icon="carrot-down"
          height="14"
          name="CarrotDown"
          width="10"
        />
      </div>
    </div>
    <div
      class="css-1jnxvbp AlignmentOption AlignmentOption--light"
      data-test-alignment-area-option="topRight"
      data-test-box=""
    >
      <div
        class="css-1jnxvbp AlignmentOption__optionView AlignmentOption__optionView--light AlignmentOption__optionView--topRight AlignmentOption__optionView--topRight--light"
        data-test-box=""
      >
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
      </div>
    </div>
    <div
      class="Tooltip Tooltip--light"
    >
      <div
        class="Tooltip__content Tooltip__content--light"
      >
        Align top right
        <div
          class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
          data-test-icon="carrot-down"
          height="14"
          name="CarrotDown"
          width="10"
        />
      </div>
    </div>
    <div
      class="css-1jnxvbp AlignmentOption AlignmentOption--light"
      data-test-alignment-area-option="left"
      data-test-box=""
    >
      <div
        class="css-1jnxvbp AlignmentOption__optionView AlignmentOption__optionView--light AlignmentOption__optionView--left AlignmentOption__optionView--left--light"
        data-test-box=""
      >
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
      </div>
    </div>
    <div
      class="Tooltip Tooltip--light"
    >
      <div
        class="Tooltip__content Tooltip__content--light"
      >
        Align left
        <div
          class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
          data-test-icon="carrot-down"
          height="14"
          name="CarrotDown"
          width="10"
        />
      </div>
    </div>
    <div
      class="css-1jnxvbp AlignmentOption AlignmentOption--light"
      data-test-alignment-area-option="center"
      data-test-box=""
    >
      <div
        class="css-1jnxvbp AlignmentOption__optionView AlignmentOption__optionView--light AlignmentOption__optionView--center AlignmentOption__optionView--center--light"
        data-test-box=""
      >
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
      </div>
    </div>
    <div
      class="Tooltip Tooltip--light"
    >
      <div
        class="Tooltip__content Tooltip__content--light"
      >
        Align center
        <div
          class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
          data-test-icon="carrot-down"
          height="14"
          name="CarrotDown"
          width="10"
        />
      </div>
    </div>
    <div
      class="css-1jnxvbp AlignmentOption AlignmentOption--light"
      data-test-alignment-area-option="right"
      data-test-box=""
    >
      <div
        class="css-1jnxvbp AlignmentOption__optionView AlignmentOption__optionView--light AlignmentOption__optionView--right AlignmentOption__optionView--right--light"
        data-test-box=""
      >
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
      </div>
    </div>
    <div
      class="Tooltip Tooltip--light"
    >
      <div
        class="Tooltip__content Tooltip__content--light"
      >
        Align right
        <div
          class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
          data-test-icon="carrot-down"
          height="14"
          name="CarrotDown"
          width="10"
        />
      </div>
    </div>
    <div
      class="css-1jnxvbp AlignmentOption AlignmentOption--light"
      data-test-alignment-area-option="bottomLeft"
      data-test-box=""
    >
      <div
        class="css-1jnxvbp AlignmentOption__optionView AlignmentOption__optionView--light AlignmentOption__optionView--bottomLeft AlignmentOption__optionView--bottomLeft--light"
        data-test-box=""
      >
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
      </div>
    </div>
    <div
      class="Tooltip Tooltip--light"
    >
      <div
        class="Tooltip__content Tooltip__content--light"
      >
        Align bottom left
        <div
          class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
          data-test-icon="carrot-down"
          height="14"
          name="CarrotDown"
          width="10"
        />
      </div>
    </div>
    <div
      class="css-1jnxvbp AlignmentOption AlignmentOption--light AlignmentOption--selected AlignmentOption--selected--light"
      data-test-alignment-area-option="bottomCenter"
      data-test-box=""
    >
      <div
        class="css-1jnxvbp AlignmentOption__optionView AlignmentOption__optionView--light AlignmentOption__optionView--bottomCenter AlignmentOption__optionView--bottomCenter--light"
        data-test-box=""
      >
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light AlignmentOption__indicator--selected AlignmentOption__indicator--selected--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light AlignmentOption__indicator--selected AlignmentOption__indicator--selected--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light AlignmentOption__indicator--selected AlignmentOption__indicator--selected--light"
          data-test-box=""
        />
      </div>
    </div>
    <div
      class="Tooltip Tooltip--light"
    >
      <div
        class="Tooltip__content Tooltip__content--light"
      >
        Align bottom center
        <div
          class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
          data-test-icon="carrot-down"
          height="14"
          name="CarrotDown"
          width="10"
        />
      </div>
    </div>
    <div
      class="css-1jnxvbp AlignmentOption AlignmentOption--light"
      data-test-alignment-area-option="bottomRight"
      data-test-box=""
    >
      <div
        class="css-1jnxvbp AlignmentOption__optionView AlignmentOption__optionView--light AlignmentOption__optionView--bottomRight AlignmentOption__optionView--bottomRight--light"
        data-test-box=""
      >
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
        <div
          class="css-1jnxvbp AlignmentOption__indicator AlignmentOption__indicator--light"
          data-test-box=""
        />
      </div>
    </div>
    <div
      class="Tooltip Tooltip--light"
    >
      <div
        class="Tooltip__content Tooltip__content--light"
      >
        Align bottom right
        <div
          class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
          data-test-icon="carrot-down"
          height="14"
          name="CarrotDown"
          width="10"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
